### PR TITLE
Shutdown server on ACTION_POWER_DISCONNECT

### PR DESCRIPTION
**Pull request by vmaxim**
Shutdown the server through the broadcast receiver on [ACTION_POWER_DISCONNECTED](https://developer.android.com/reference/android/content/Intent.html#ACTION_POWER_DISCONNECTED). 
This behaviour can be controlled using `shutdownOnPowerDisconnect` setting whose value is `true` by default.

Addressed to: appium/appium#9412
CI results: https://travis-ci.org/vmaxim/appium-uiautomator2-server/builds/371962846

### CODE REVIEW

File: app/src/main/java/io/appium/uiautomator2/server/ServerInstrumentation.java [hunk 1]
```diff
@@ -1,42 +1,82 @@
...
     private static ServerInstrumentation instance;
-    private static Context context;
+
+    private final Context context;
     private HttpdThread serverThread;
     private PowerManager.WakeLock wakeLock;
-    private int serverPort;
-    private  boolean isStopServer;
+    private final int serverPort;
+    private boolean isStopServer;
```
[the remark this answers is not in the document]
Comment (vmaxim) [reply app/src/main/java/io/appium/uiautomator2/server/ServerInstrumentation.java]: I would prefer not to change the current start/stop logic in this `PR`.
If you take a look closely you will see that the flow is not so trivial. I hope I could do this in separate PR.

[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply app/src/main/java/io/appium/uiautomator2/server/ServerInstrumentation.java]: it is only about giving a better name to this variable. You don't have to change any logic

**Comment by KnightRadiantZ** [comment]
@mykola-mokhnach i tried scouring the net before i post this and i did not find any so i am asking here, how can i set shutdownOnPowerDisconnect to false? is it through Appium cap?

**Comment by mykola-mokhnach** [comment]
This could be done via settings interface. Check https://github.com/appium/java-client/blob/1991a8a0f9e4a3ff467dbb713cb5c51c8edc060f/src/test/java/io/appium/java_client/android/SettingTest.java

**Pull request merged**